sql/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java

```diff
@@ -124,6 +125,28 @@ public void testSubscriptSplitPatternMatcher() throws Exception {
     }
 
     @Test
+    public void testAnalyzeSubscriptFunctionCall() throws Exception {
+        // Test that when operating on a table with generated columns that
```

Comment:
has nothing to do with generated columns, it's a generic subscript issue. pls update the comment